lib/store_model/types/polymorphic_array_type.rb

```diff
@@ -0,0 +1,87 @@
+# frozen_string_literal: true
+
+require "active_model"
+
+module StoreModel
+  module Types
+    # Implements ActiveModel::Type::Value type for handling an array of
+    # StoreModel::Model
+    class PolymorphicArrayType < ActiveModel::Type::Value
```

Comment:
This class look very similar to the old `ArrayType`, maybe it would be reasonable to define a superclass for them (but let's do it later, when everything else is handled)